### PR TITLE
Add unit tests for persistence task types in DataManagerInterfaces

### PR DESCRIPTION
**Pull request by timl3136**
**What changed?**
Add unit tests for some task types used by persistence and some other functions

**Why?**
Improve unit test coverage and stability of the platform

**How did you test it?**
unit test

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: common/persistence/dataManagerInterfaces_test.go [hunk 1]
```diff
@@ -22,7 +22,9 @@ package persistence
 
 import (
 	"errors"
+	"fmt"
```
Comment (taylanisikdemir) [inline common/persistence/dataManagerInterfaces_test.go:25]: let's rename the test file and corresponding go file to camel_case

File: common/persistence/dataManagerInterfaces_test.go [hunk 2]
```diff
@@ -58,3 +60,65 @@ func TestIsTransientError(t *testing.T) {
...
+		&RecordWorkflowStartedTask{Version: 1, TaskID: 1, VisibilityTimestamp: timeNow},
+		&ResetWorkflowTask{Version: 1, TaskID: 1, VisibilityTimestamp: timeNow},
+		&CloseExecutionTask{Version: 1, TaskID: 1, VisibilityTimestamp: timeNow},
+		&DeleteHistoryEventTask{Version: 1, TaskID: 1, VisibilityTimestamp: timeNow},
+		&DecisionTimeoutTask{Version: 1, TaskID: 1, VisibilityTimestamp: timeNow},
+		&ActivityTimeoutTask{Version: 1, TaskID: 1, VisibilityTimestamp: timeNow},
+		&UserTimerTask{Version: 1, TaskID: 1, VisibilityTimestamp: timeNow},
+	}
+
+	for _, task := range tasks {
```
Comment (taylanisikdemir) [inline common/persistence/dataManagerInterfaces_test.go:84]: you can define `tasks` as a map to co-locate object and its expected type

Comment (timl3136) [reply common/persistence/dataManagerInterfaces_test.go:84]: Given that transfer task and non-transfer task share the same set of int keys, I think we should keep the current way of testing.

File: common/persistence/dataManagerInterfaces_test.go (hunk 1 above)
Comment (timl3136) [reply common/persistence/dataManagerInterfaces_test.go]: Renamed.

**Review by taylanisikdemir — APPROVED** [review]

**Pull request merged**